src/rust/src/backend/keys.rs

```diff
@@ -34,7 +34,12 @@ pub(crate) fn load_der_private_key_bytes<'p>(
     password: Option<&[u8]>,
     unsafe_skip_rsa_key_validation: bool,
 ) -> CryptographyResult<pyo3::Bound<'p, pyo3::PyAny>> {
-    if let Ok(pkey) = openssl::pkey::PKey::private_key_from_der(data) {
+    let pkey = cryptography_key_parsing::pkcs8::parse_private_key(data)
```

Comment:
This nit doesn't really matter, but it seems probable that the least likely key type is DSA so having it be the first fallback is likely inefficient.